ssh-keygen.c

```diff
@@ -1541,12 +1541,9 @@ do_change_comment(struct passwd *pw)
 	sshkey_free(private);
 
 	strlcat(identity_file, ".pub", sizeof(identity_file));
-	fd = open(identity_file, O_WRONLY | O_CREAT | O_TRUNC, 0644);
-	if (fd == -1)
-		fatal("Could not save your public key in %s", identity_file);
-	f = fdopen(fd, "w");
+	f = fopen(identity_file, "w");
 	if (f == NULL)
-		fatal("fdopen %s failed: %s", identity_file, strerror(errno));
+		fatal("fopen %s failed: %s", identity_file, strerror(errno));
```

Comment:
We cannot change core OpenSSH code. Add a #ifdef block.. Look elsewhere in this file that also had to deal with fdopen.